cli/cli/Services/DiscoveryService.cs

```diff
@@ -43,25 +43,51 @@ public async IAsyncEnumerable<ServiceDiscoveryEvent> StartDiscovery(TimeSpan tim
 
 
 		// This doesn't actually block
-		await _localSystem.StartListeningToDockerRaw((beamoId, eventType, raw) =>
+		await _localSystem.StartListeningToDockerRaw(async (beamoId, eventType, raw) =>
 		{
-			// We skip out on non-Microservice containers.
+			if (eventType != "start" && eventType != "destroy")
+			{
+				return;
+			}
+			
+			
+			var isRunning = eventType == "start";
+			
 			var serviceDefinition = _localSystem.BeamoManifest.ServiceDefinitions.FirstOrDefault(sd => sd.BeamoId == beamoId);
-			if (serviceDefinition == null || serviceDefinition.Protocol != BeamoProtocolType.HttpMicroservice) return;
+			if (serviceDefinition == null) return;
 
-			var protocol = _localSystem.BeamoManifest.HttpMicroserviceRemoteProtocols[serviceDefinition.BeamoId];
-			var isRunning = eventType == "start";
-			isRunning &= eventType != "stop";
-			var evt = CreateEvent(new ServiceDiscoveryEntry()
+			var healthPort = 0;
+
+			if (serviceDefinition.Protocol == BeamoProtocolType.HttpMicroservice)
+			{
+				healthPort = Convert.ToInt32(_localSystem.BeamoManifest.HttpMicroserviceRemoteProtocols[serviceDefinition.BeamoId]
+					.HealthCheckPort);
+			}
+			else
+			{
+				try
+				{
+					var port = await _localSystem.GetStorageHostPort(beamoId);
+					healthPort = Convert.ToInt32(port);
+				}
+				catch
+				{
+					// storage is not running, therefore there is no port available
+				}
+			}
+
+			var service = new ServiceDiscoveryEntry()
 			{
 				cid = _appContext.Cid,
 				pid = _appContext.Pid,
 				prefix = MachineHelper.GetUniqueDeviceId(),
 				serviceName = serviceDefinition.BeamoId,
-				healthPort = Convert.ToInt32(protocol.HealthCheckPort),
+				healthPort = healthPort,

```

Comment:
I don't love that the `healthPort` is being used to describe the _data_ port for a service instance. 

I think I'd prefer to use more of a union type approach, where the `ServiceDiscoveryEntry` gets two new fields, 
```
public string serviceType;
public int dataPort;
```

And the `serviceType` gets set to either `service` or `storage`, and then depending on that, either the `healthPort` OR the `dataPort` gets set.

Reply:
will do!